components/properties/src/props.rs

```diff
@@ -41,6 +49,147 @@ enum EnumeratedProperty {
+    /// names that match exactly
+    #[inline]
+    pub fn get_strict_u16(&self, name: &str) -> Option<u16> {
+        get_strict_u16(&self.map, name)
+    }
+
+    /// Get the property value as a `T`, doing a strict search looking for
+    /// names that match exactly
+    #[inline]
+    pub fn get_strict(&self, name: &str) -> Option<T> {
+        self.get_strict_u16(name)
+            .and_then(|u| T::try_from_u32(u as u32).ok())
+    }
+
+    /// Get the property value as a u16, doing a loose search looking for
+    /// names that match case-insensitively, ignoring ASCII hyphens, underscores, and
+    /// whitespaces.
+    #[inline]
+    pub fn get_loose_u16(&self, name: &str) -> Option<u16> {
+        get_loose_u16(&self.map, name)
+    }
+
+    /// Get the property value as a `T`, doing a loose search looking for
+    /// names that match case-insensitively, ignoring ASCII hyphens, underscores, and
+    /// whitespaces.
+    #[inline]
+    pub fn get_loose(&self, name: &str) -> Option<T> {
+        get_loose_u16(&self.map, name).and_then(|u| T::try_from_u32(u as u32).ok())
+    }
+
+    /// Construct a new one from loaded data
+    ///
+    /// Typically it is preferable to use methods on individual property value types
+    /// (like [`Script::lookup_values()`]) instead.
+    pub fn from_data<M>(data: DataPayload<M>) -> Self
+    where
+        M: DataMarker<Yokeable = PropertyValueNameMapV1<'static>>,
+    {
+        Self {
+            map: data.map_project(|m, _| m),
+            marker: PhantomData,
+        }
+    }
+}
+
+/// Avoid monomorphizing multiple copies of this function
+fn get_strict_u16(map: &DataPayload<ErasedValueNameMap>, name: &str) -> Option<u16> {
+    // NormalizedPropertyName has no invariants so this should be free, but
+    // avoid introducing a panic regardless
+    let name = NormalizedPropertyName::parse_byte_slice(name.as_bytes()).ok()?;
+    map.get().map.get_copied_by(|p| {
+        let cmp = p.cmp(name);
+        // For strict matching, use the same comparator but
+        // in the Equal case make sure to actually check for full equality
+        // (implemented by Eq, as opposed to the Ord impl)
+        if cmp == Ordering::Equal {
+            if p == name {
+                Ordering::Equal
+            } else {
+                // There's no way to signal "unequal, just stop now",
+                // so we pretend that invalid names are always less than the valid ones
+                Ordering::Greater
+            }
+        } else {
+            cmp
+        }
+    })
+}
+
+/// Avoid monomorphizing multiple copies of this function
+fn get_loose_u16(map: &DataPayload<ErasedValueNameMap>, name: &str) -> Option<u16> {
+    // NormalizedPropertyName has no invariants so this should be free, but
+    // avoid introducing a panic regardless
+    let name = NormalizedPropertyName::parse_byte_slice(name.as_bytes()).ok()?;
+    map.get().map.get_copied(name)
+}
+
+macro_rules! impl_value_getter {
```

Comment:
while we can't reduce _much_ of the open enum boilerplate, I'm wondering if we should move the rest of the enum into this macro so it at least implements the right traits and such uniformly

probably for a followup